### DISCUSSION TITLE
Using iota.c lib for communicating with smart contract running on node

### DISCUSSION THREAD

**Discussion by mahesh2392** [iotaledger/iota.c#461] [Q&A] [answered]
I have few questions regarding iota c lib as we are developing on embedded application which uses iota c lib...
  1.Is it possible to communicate with deployed smart contract on node with this iota c library? is iota c lib capable to communicate 
     with smart contract?
  2.I am running wasp node on my PC and deploying contract on wasp node. so is it possible to communicate with wasp node with 
     this  library? Is it possible to run apis like get node info so I can get my actual wasp node information or invoke any smart           
     contract functions? 
  3.if yes how we can do it?

Thanks...Any help really appreciate...

**Comment by oopsmonk** [comment, marked as answer]
Hi @mahesh2392 currently, the WASP node communicates with the goshimmer node at the moment, this library is not compatible with goShimmer APIs. It might be possible to use this library to do smart-contracts-related applications via node APIs in the feature if the WASP can use Hornet APIs.

**Reply by mahesh2392** [reply]
Hello @oopsmonk Thanks for reply.... so as per current IOTA C lib can communicate only with IOTA node(Hornet)...is there any possibility that by using WASP Public APIs (https://api.wasp.sc.iota.org/doc#/) we can communicate with smart contract via IOTA C library ??
As I am using wasp-cli to deploy chain and contract as well as communication with contract via WASP Public APIs but what I see for every POST and GET wasp-cli wallets IOTA balance get deducted (e.g. Posted on-ledger transaction 4y3aCfUSyiTgJ7osBMopezrTvkJRm7LrH6QL4rvFqkZh) thats mean we need IOTA funds to make call to contract.
Whats your thoughts Is there any possibility with current IOTA C library to perform such activities like WASP-CLI or any workaround ?? Thanks again!!!!

**Reply by oopsmonk** [reply]
You need a wasp-cli in C language. wasp-cli is in the early stage and I think it's not good to maintain a client lib in different programming languages.
